### DISCUSSION TITLE
GUI Layout

### DISCUSSION THREAD

**Discussion by scottctd** [UofT-CS-Group/CSC110-Final-Project#12] [General]
- What will Mario want to see?

**Comment by scottctd** [comment]
# Widgets and Descriptions

- I divided those widgets by their functionalities.
- In my opinion, they should also be put into **groups** in our GUI, instead of spreading out randomly.

![image](https://user-images.githubusercontent.com/35100768/142749680-3b75fb57-4e06-4644-aa73-7b0312ca516a.png)

## Decorative

- Charlotte's icon is very beautiful and we could put it on the main GUI.
- (Optional) Title (Or we could just set the window title to our title)
    - To be honest, I don't like there is a big title in our program, that seems very odd and unfashionable.

## Location Selection

- Country Selection Label
- Country Selection Combobox
- Country Search Bar
    - Use to search for a country.
- Country Shortcuts Buttons for countries like China, US, Canada, ...
- Reset Country Push Button

We could remove city and province selection and their corresponding data structures because they are actually useless.

## Date Selection

- Start Date Label
- Start Date Date Edit
- Optional: Start Date Slider
    - User could slide the bar to change the date, which is much more convenient.
- End Date Label
- End Date Date Edit
- Reset Date Push Button
- Optional: End Date Slider

## Plot

- Navigation Tool Bar
- Figure

## Settings

- If the remaining space is large (I suggest making our buttons and labels compact, not spreading out), then we could put some settings like line color, font, or even language selections.
- These are very open and we could make almost everything configurable if we want.
- If the remaining space is just good, we could let our setting go into the menu bar.

## Table View

- If we could make it, we should let the table view be the same position as our plots, and use something just like the graph below to select between plot and table.

![image](https://user-images.githubusercontent.com/35100768/142749787-28eb6a9e-85cd-44e8-b955-5620a6d5d85b.png)

- Table Search bar
    - There gonna be a complex algorithm to let user only need to search onlythe key words of our data. 
- Filter
    - This is really advanced, but we may let user to write some logic (predicate) to filter our data
    - Let our ta do coding on our program lollll
- More... (Table view is WIP so for now, we could just consider our other widgets)

# Last but not least,

- I may incorporate the init window into our main window.
- Let the status bar display a progress bar indicating our init progress is much better maybe.

**Comment by scottctd** [comment]
My first draft:

![image](https://user-images.githubusercontent.com/35100768/142902098-d5f5479d-5ad8-4432-936b-943ada73158f.png)

**Reply by CharlotteChenyy** [reply]
我突然觉得data slider是不是有点鸡肋 因为这个slider不会太长 假如用slider来设置时间的话感觉不能弄得很准确 还不如直接在左边的框框里敲时间呢我觉得?

**Reply by scottctd** [reply]
哈哈哈这个slider是供ta玩耍的，目的不是精确（其实咱的data时间精确了也不太有用，是为了方便调时间，敲时间有的时候会挺麻烦

这样的话合理吗

不过这个逻辑可能会有点难写，因为end不能大于start

**Reply by CharlotteChenyy** [reply]
哈哈哈哈哈哈原来如此

**Reply by CharlotteChenyy** [reply]
你现在正在搞这个吗 做这个layout?

**Reply by scottctd** [reply]
没呢 我现在在refactor整个project，先画画layout提提意见，我refactor完了再写新layout和别的功能

我可能今晚就能refactor完然后pr，然后你看看我有啥不好的，明天我改完之后咱就可以用新的架构写layout了

因为现在代码越写越多全挤在一个文件很不好

**Reply by CharlotteChenyy** [reply]
好的没问题

**Reply by scottctd** [reply]
另外你的Icon有透明背景的吗

这个大白背景有点难受哈哈哈哈哈

**Reply by CharlotteChenyy** [reply]
我下了数学课 把透明背景的发到咱们微信里吧? 我操作一下哈哈哈哈哈哈
